src/sentry/web/frontend/accept_project_transfer.py

```diff
@@ -0,0 +1,110 @@
+        for o in OrganizationMember.objects.filter(user__email=request.user):
+            org_name = Organization.objects.get(id=o.organization_id).name
+            for t in o.get_teams():
+                option = " %s - %s" % (t.name, org_name)
+                teams.append([t.id, option])
+
+        self.fields['team'].choices = teams
+        self.fields['team'].widget.choices = self.fields['team'].choices
+
+
+class AcceptProjectTransferView(BaseView):
+    required_scope = 'org:admin'
+    sudo_required = True
+
+    def get_form(self, request):
+        if request.method == 'POST':
+            return AcceptProjectTransferForm(request, request.POST, initial=request.POST)
+        return AcceptProjectTransferForm(request)
+
+    def handle(self, request, *args, **kwargs):
+        try:
+            d = request.GET['data']
+        except KeyError:
+            raise Http404
+
+        try:
+            data = unsign(force_str(d))
+        except BadSignature:
+            messages.add_message(
+                request, messages.ERROR,
+                _(u'Could not approve transfer, please make sure link is valid.')
+            )
+            return HttpResponseRedirect(
+                reverse('sentry')
+            )
+
+        try:
+            data = unsign(force_str(d))
+        except SignatureExpired:
+            messages.add_message(
+                request, messages.ERROR,
+                _(u'Project transfer link has expired!')
+            )
+            return HttpResponseRedirect(
+                reverse('sentry')
+            )
+
+        project_id = data['project_id']
+        user_id = data['user_id']
+        transaction_id = data['transaction_id']
+
+        # check if user is still an owner
+        if not OrganizationMember.objects.filter(
+            role=roles.get_top_dog().id,
+            user__is_active=True,
+            user_id=user_id,
+        ).exists():
+            return HttpResponseRedirect(
+                reverse('sentry')
+            )
+
+        form = self.get_form(request)
+        if form.is_valid():
+            # transfer the project
+            team_id = form.cleaned_data.get('team')
+            new_team = Team.objects.get(id=team_id)
+            project = Project.objects.get(id=project_id)
+            project.team = new_team
+            project.organization = new_team.organization
+            project.save()
+
+            self.create_audit_entry(
+                request=request,
+                organization=project.organization,
+                target_object=project_id,
+                event=AuditLogEntryEvent.PROJECT_ACCEPT_TRANSFER,
+                data=project.get_audit_log_data(),
+                transaction_id=transaction_id,
+            )
+
+            return HttpResponseRedirect(
+                reverse('sentry-organization-home', args=[new_team.organization.slug])
+            )
+
+        context = {
+            'project': Project.objects.get(id=project_id),
```

Comment:
Move this up above L84 since you query it in both places. So we can just make it so it's queried once and just reference it down here.

Reply:
We also need to use the `from_organization_id` value in this query.

Something like:  `Project.objects.get(id=project_id, organization_id=from_organization_id)`

This will prevent the transfer from happening twice.

Reply:
And with that said, we should handle the `Project.DoesNotExist` when that happens.